resources/js/components/ui/calendar/FullCalendar.tsx

```diff
@@ -0,0 +1,266 @@
+  function setDaysSelectionAndMonth(range: DaysRange) {
+    if (tinyCalVisibleRange().intersects(daysSelection())) {
+      setTinyCalMonth(range.center());
+    }
+    setDaysSelection(range);
+  }
+  function setDaysSelectionAndMonthFromDay(day: DateTime) {
+    setDaysSelectionAndMonth(getRange(day));
+  }
+  function moveDaysSelection(dir: 1 | -1) {
+    // Use end of range to preserve week type.
+    setDaysSelectionAndMonthFromDay(daysSelection().end.plus({[mode()]: dir}));
+  }
+
+  // Set the days selection when the mode is changed.
+  createComputed(
+    on(mode, (mode, prevMode) => {
+      if (prevMode && mode !== prevMode) {
+        // Restore the previous selection for this mode if it overlaps with the current selection.
+        const savedDaysSelection = daysSelectionByMode.get(mode);
+        if (savedDaysSelection?.intersects(daysSelection())) {
+          setDaysSelectionAndMonth(savedDaysSelection);
+        } else {
+          // Calculate the new selection from the old selection (from different mode).
+          if (mode === "month") {
+            setDaysSelectionAndMonthFromDay(daysSelection().center());
+          } else if (mode === "week") {
+            if (prevMode === "month") {
+              // Never change tinyCalMonth when switching from month to week.
+              if (daysSelectionByMode.get("week")?.length() === 7) {
+                // Select a calendar week.
+                setDaysSelection(weekDayCalculator().dayToWeek(daysSelection().start));
+              } else {
+                // Select the first work week of this month.
+                for (const day of daysSelection()) {
+                  if (!weekDayCalculator().isWeekend(day)) {
+                    setDaysSelection(weekDayCalculator().dayToWorkdays(day));
+                    break;
+                  }
+                }
+              }
+            } else {
+              setDaysSelectionAndMonthFromDay(daysSelection().start);
+            }
+          } else if (mode === "day") {
+            setDaysSelectionAndMonthFromDay(daysSelection().start);
+          } else {
+            return mode satisfies never;
+          }
+        }
+      }
+    }),
+  );
+  // Set the resources selection when the resources selection mode is changed.
+  createComputed(
+    on(resourcesSelectionMode, (mode, prevMode) => {
+      if (prevMode && mode !== prevMode) {
+        // Restore the previous selection for this mode if it overlaps with the current selection.
+        const savedResourcesSelection = resourcesSelectionByMode[mode];
+        if (savedResourcesSelection?.some((id) => selectedResources().includes(id))) {
+          setSelectedResources(savedResourcesSelection);
+        } else {
+          if (mode === "radio") {
+            setSelectedResources(selectedResources().slice(0, 1));
+          } else if (mode === "checkbox") {
+            // Use the stored selections because `selectedProps()` might be already changed to the new type.
```

Comment:
s/selectedProps/selectedResources ?

a wyżej to nie przeszkadza, że might be already changed (or not)?